setup.py

```diff
@@ -17,11 +19,194 @@
 #  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 #  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 #  DEALINGS IN THE SOFTWARE.
+"""Molecule distribution package setuptools installer."""
 
 import setuptools
+import setuptools.command.dist_info
 
-setuptools.setup(
-    packages=setuptools.find_packages(),
-    pbr=True,
-    setup_requires=['pbr']
-)
+
+ALL_STRING_TYPES = tuple(map(type, ('', b'', u'')))
+
+
+def str_if_nested_or_str(s):
+    """Turn input into a native string if possible."""
+    if isinstance(s, ALL_STRING_TYPES):
+        return str(s)
+    if isinstance(s, (list, tuple)):
+        return type(s)(map(str_if_nested_or_str, s))
+    if isinstance(s, (dict, )):
+        return stringify_dict_contents(s)
+    return s
+
+
+def stringify_dict_contents(dct):
+    """Turn dict keys and values into native strings."""
+    return {
+        str_if_nested_or_str(k): str_if_nested_or_str(v)
+        for k, v in dct.items()
+    }
+
+
+try:
+    from setuptools.config import read_configuration, ConfigOptionsHandler
+    import setuptools.config
+    import setuptools.dist
+
+    # Set default value for 'use_scm_version'
+    setattr(setuptools.dist.Distribution, 'use_scm_version', False)
+
+    # Attach bool parser to 'use_scm_version' option
+    class ShimConfigOptionsHandler(ConfigOptionsHandler):
+        """Extension class for ConfigOptionsHandler."""
+
+        @property
+        def parsers(self):
+            """Return an option mapping with default data type parsers."""
+            _orig_parsers = super(ShimConfigOptionsHandler, self).parsers
+            return dict(use_scm_version=self._parse_bool, **_orig_parsers)
+
+        def parse_section_packages__find(self, section_options):
```

Comment:
```suggestion
        def parse_section_packages_find(self, section_options):
```

?

Reply:
No, this must match the parent class:
https://github.com/pypa/setuptools/blob/7db0102/setuptools/config.py#L587-L606